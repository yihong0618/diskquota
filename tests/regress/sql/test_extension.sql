-- NOTE: when test this script, you must make sure that there is no diskquota
-- worker process.
CREATE DATABASE dbx0 ;
CREATE DATABASE dbx1 ;
CREATE DATABASE dbx2 ;
CREATE DATABASE dbx3 ;
CREATE DATABASE dbx4 ;
CREATE DATABASE dbx5 ;
CREATE DATABASE dbx6 ;
CREATE DATABASE dbx7 ;
CREATE DATABASE dbx8 ;
CREATE DATABASE dbx9 ;
CREATE DATABASE dbx10 ;

--start_ignore
\! gpconfig -c diskquota.max_workers -v 20 --skipvalidation
\! gpstop -arf
--end_ignore
\c
show max_worker_processes;
show diskquota.max_workers;

SELECT diskquota.wait_for_worker_new_epoch();

\c dbx0
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx1
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
INSERT INTO SX.a values(generate_series(0, 100000));
CREATE EXTENSION diskquota;
SELECT diskquota.init_table_size_table();
SELECT diskquota.wait_for_worker_new_epoch();
SELECT diskquota.set_schema_quota('SX', '1MB');
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx2
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx3
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx4
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx5
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx6
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx7
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx8
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx9
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();

\c dbx10
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();

\c dbx0
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c dbx1
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c dbx2
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c dbx3
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c dbx4
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c dbx5
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c dbx6
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c dbx7
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c dbx8
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c dbx9
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c dbx10
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c contrib_regression

DROP DATABASE dbx0 ;
DROP DATABASE dbx1 ;
DROP DATABASE dbx2 ;
DROP DATABASE dbx3 ;
DROP DATABASE dbx4 ;
DROP DATABASE dbx5 ;
DROP DATABASE dbx6 ;
DROP DATABASE dbx7 ;
DROP DATABASE dbx8 ;
DROP DATABASE dbx9 ;
DROP DATABASE dbx10 ;
--start_ignore
\! gpconfig -c diskquota.max_workers -v 1 --skipvalidation
\! gpstop -arf;
--end_ignore
\c
show diskquota.max_workers;
