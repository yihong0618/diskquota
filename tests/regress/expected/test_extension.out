-- NOTE: when test this script, you must make sure that there is no diskquota
-- worker process.
CREATE DATABASE dbx0 ;
CREATE DATABASE dbx1 ;
CREATE DATABASE dbx2 ;
CREATE DATABASE dbx3 ;
CREATE DATABASE dbx4 ;
CREATE DATABASE dbx5 ;
CREATE DATABASE dbx6 ;
CREATE DATABASE dbx7 ;
CREATE DATABASE dbx8 ;
CREATE DATABASE dbx9 ;
CREATE DATABASE dbx10 ;
--start_ignore
\! gpconfig -c diskquota.max_workers -v 20 --skipvalidation
20220802:15:47:27:028366 gpconfig:wxiaoran-a01:xiwang-[INFO]:-completed successfully with parameters '-c diskquota.max_workers -v 20 --skipvalidation'
\! gpstop -arf
20220802:16:43:25:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Starting gpstop with args: -arf
20220802:16:43:25:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Gathering information and validating the environment...
20220802:16:43:25:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Obtaining Greenplum Master catalog information
20220802:16:43:25:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Obtaining Segment details from master...
20220802:16:43:25:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Greenplum Version: 'postgres (Greenplum Database) 6.20.3+dev.5.g4bc90eab02 build dev'
20220802:16:43:25:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Commencing Master instance shutdown with mode='fast'
20220802:16:43:25:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Master segment instance directory=/Users/xiwang/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1/
20220802:16:43:25:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Attempting forceful termination of any leftover master process
20220802:16:43:25:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Terminating processes for segment /Users/xiwang/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1/
20220802:16:43:25:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Stopping master standby host wxiaoran-a01.vmware.com mode=fast
20220802:16:43:26:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Successfully shutdown standby process on wxiaoran-a01.vmware.com
20220802:16:43:26:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Targeting dbid [2, 5, 3, 6, 4, 7] for shutdown
20220802:16:43:26:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Commencing parallel primary segment instance shutdown, please wait...
20220802:16:43:26:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-0.00% of jobs completed
20220802:16:43:28:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-100.00% of jobs completed
20220802:16:43:28:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Commencing parallel mirror segment instance shutdown, please wait...
20220802:16:43:28:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-0.00% of jobs completed
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-100.00% of jobs completed
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-----------------------------------------------------
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-   Segments stopped successfully      = 6
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-   Segments with errors during stop   = 0
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-----------------------------------------------------
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Successfully shutdown 6 of 6 segment instances 
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Database successfully shutdown with no errors reported
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Cleaning up leftover gpmmon process
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-No leftover gpmmon process found
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Cleaning up leftover gpsmon processes
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-No leftover gpsmon processes on some hosts. not attempting forceful termination on these hosts
20220802:16:43:29:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Cleaning up leftover shared memory
20220802:16:43:30:058210 gpstop:wxiaoran-a01:xiwang-[INFO]:-Restarting System...
--end_ignore
\c
show max_worker_processes;
 max_worker_processes 
----------------------
 20
(1 row)

show diskquota.max_workers;
 diskquota.max_workers 
-----------------------
 20
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

\c dbx0
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name: sx
DROP TABLE SX.a;
\c dbx1
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
INSERT INTO SX.a values(generate_series(0, 100000));
CREATE EXTENSION diskquota;
WARNING:  [diskquota] diskquota is not ready because current database is not empty
HINT:  please run 'SELECT diskquota.init_table_size_table();' to initialize diskquota
SELECT diskquota.init_table_size_table();
 init_table_size_table 
-----------------------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name: sx
DROP TABLE SX.a;
\c dbx2
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name: sx
DROP TABLE SX.a;
\c dbx3
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name: sx
DROP TABLE SX.a;
\c dbx4
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name: sx
DROP TABLE SX.a;
\c dbx5
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name: sx
DROP TABLE SX.a;
\c dbx6
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name: sx
DROP TABLE SX.a;
\c dbx7
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name: sx
DROP TABLE SX.a;
\c dbx8
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

CREATE SCHEMA SX;
CREATE TABLE SX.a(i int) DISTRIBUTED BY (i);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000));
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name: sx
DROP TABLE SX.a;
\c dbx9
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

\c dbx10
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

\c dbx0
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c dbx1
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c dbx2
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c dbx3
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c dbx4
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c dbx5
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c dbx6
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c dbx7
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c dbx8
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c dbx9
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c dbx10
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c contrib_regression
DROP DATABASE dbx0 ;
DROP DATABASE dbx1 ;
DROP DATABASE dbx2 ;
DROP DATABASE dbx3 ;
DROP DATABASE dbx4 ;
DROP DATABASE dbx5 ;
DROP DATABASE dbx6 ;
DROP DATABASE dbx7 ;
DROP DATABASE dbx8 ;
DROP DATABASE dbx9 ;
DROP DATABASE dbx10 ;
--start_ignore
\! gpconfig -c diskquota.max_workers -v 1 --skipvalidation
20220802:15:49:09:029439 gpconfig:wxiaoran-a01:xiwang-[INFO]:-completed successfully with parameters '-c diskquota.max_workers -v 1 --skipvalidation'
\! gpstop -arf;
20220802:16:32:34:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Starting gpstop with args: -arf
20220802:16:32:34:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Gathering information and validating the environment...
20220802:16:32:34:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Obtaining Greenplum Master catalog information
20220802:16:32:34:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Obtaining Segment details from master...
20220802:16:32:34:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Greenplum Version: 'postgres (Greenplum Database) 6.20.3+dev.5.g4bc90eab02 build dev'
20220802:16:32:34:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Commencing Master instance shutdown with mode='fast'
20220802:16:32:34:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Master segment instance directory=/Users/xiwang/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1/
20220802:16:32:35:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Attempting forceful termination of any leftover master process
20220802:16:32:35:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Terminating processes for segment /Users/xiwang/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1/
20220802:16:32:36:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Stopping master standby host wxiaoran-a01.vmware.com mode=fast
20220802:16:32:37:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Successfully shutdown standby process on wxiaoran-a01.vmware.com
20220802:16:32:37:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Targeting dbid [2, 5, 3, 6, 4, 7] for shutdown
20220802:16:32:37:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Commencing parallel primary segment instance shutdown, please wait...
20220802:16:32:37:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-0.00% of jobs completed
20220802:16:32:40:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-100.00% of jobs completed
20220802:16:32:40:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Commencing parallel mirror segment instance shutdown, please wait...
20220802:16:32:40:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-0.00% of jobs completed
20220802:16:32:41:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-100.00% of jobs completed
20220802:16:32:41:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-----------------------------------------------------
20220802:16:32:41:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-   Segments stopped successfully      = 6
20220802:16:32:41:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-   Segments with errors during stop   = 0
20220802:16:32:41:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-----------------------------------------------------
20220802:16:32:41:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Successfully shutdown 6 of 6 segment instances 
20220802:16:32:41:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Database successfully shutdown with no errors reported
20220802:16:32:41:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Cleaning up leftover gpmmon process
20220802:16:32:42:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-No leftover gpmmon process found
20220802:16:32:42:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Cleaning up leftover gpsmon processes
20220802:16:32:42:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-No leftover gpsmon processes on some hosts. not attempting forceful termination on these hosts
20220802:16:32:42:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Cleaning up leftover shared memory
20220802:16:32:44:046832 gpstop:wxiaoran-a01:xiwang-[INFO]:-Restarting System...
--end_ignore
\c
show diskquota.max_workers;
 diskquota.max_workers 
-----------------------
 1
(1 row)

